problem_builder/step_review.py

```diff
@@ -0,0 +1,291 @@
+
+    @property
+    def display_name_with_default(self):
+        return self._(self.STUDIO_LABEL)
+
+    def is_applicable(self, context):
+        """ Return true if this block should appear in the review step, false otherwise """
+        score_summary = context['score_summary']
+        attempts_remain = not score_summary['max_attempts_reached']
+        if (
+            (self.num_attempts_condition == self.IF_ATTEMPTS_REMAIN and not attempts_remain) or
+            (self.num_attempts_condition == self.IF_NO_ATTEMPTS_REMAIN and attempts_remain)
+        ):
+            return False
+
+        perfect_score = (score_summary['incorrect'] == 0 and score_summary['partial'] == 0)
+        if (
+            (self.score_condition == self.SCORE_PERFECT and not perfect_score) or
+            (self.score_condition == self.SCORE_IMPERFECT and perfect_score)
+        ):
+            return False
+
+        return True
+
+    def student_view(self, context=None):
+        """ Render this message. """
+        html = u'<div class="review-conditional-message">{content}</div>'.format(
+            content=self.content
+        )
+        return Fragment(html)
+
+    preview_view = student_view
+    mentoring_view = student_view  # Same as student_view but Studio won't wrap it with the editing header/buttons
+
+    def author_view(self, context=None):
+        fragment = self.student_view(context)
+        desc = ""
+        if self.num_attempts_condition == self.ATTEMPTS_ANY and self.score_condition == self.SCORE_ANY:
+            desc = self._("Always shown")
+        else:
+            if self.score_condition != self.SCORE_ANY:
+                desc += self.SCORE_CONDITIONS_DESCRIPTIONS[self.score_condition] + "<br>"
+            if self.num_attempts_condition != self.ATTEMPTS_ANY:
+                desc += self.NUM_ATTEMPTS_COND_DESCRIPTIONS[self.num_attempts_condition]
+        fragment.content += u'<div class="submission-message-help"><p>{}</p></div>'.format(desc)
+        return fragment
+
+
+@XBlock.needs("i18n")
+class ScoreSummaryBlock(XBlockWithTranslationServiceMixin, XBlockWithPreviewMixin, XBlock):
+    """
+    Summaryize the score that the student earned.
+    """
+    CATEGORY = 'sb-review-score'
+    STUDIO_LABEL = _("Score Summary")
+    has_author_view = True  # Without this flag, studio will use student_view on newly-added blocks :/
+
+    @property
+    def display_name_with_default(self):
+        return self._(self.STUDIO_LABEL)
+
+    def student_view(self, context=None):
+        """ Render the score summary message. """
+        html = loader.render_template("templates/html/sb-review-score.html", context.get("score_summary", {}))
+        return Fragment(html)
+
+    mentoring_view = student_view  # Same as student_view but Studio won't wrap it with the editing header/buttons
+
+    def author_view(self, context=None):
+        if not context.get("score_summary"):
```

Comment:
Same here.  (Not sure whether that ever happens, but then I wonder why we have a default value for `context` anyway.)